core/src/test/scala/org/apache/spark/util/ClosureCleanerSuite.scala

```diff
@@ -260,6 +262,21 @@ private object TestUserClosuresActuallyCleaned {
   def testMapPartitionsWithIndex(rdd: RDD[Int]): Unit = {
     rdd.mapPartitionsWithIndex { (_, it) => return; it }.count()
   }
+  def testFlatMapWith(rdd: RDD[Int]): Unit = {
+    import java.util.Random
+    val randoms = rdd.flatMapWith(
+      (index: Int) => new Random(index + 42))
+      {(t: Int, prng: Random) =>
+        val random = prng.nextDouble()
+        Seq(random * t, random * t * 10)}.
+      count()
+  }
+  def testMapWith(rdd: RDD[Int]): Unit = {
+    import java.util.Random
+    val randoms = rdd.mapWith(
+      (index: Int) => new Random(index + 42))
+      {(t: Int, prng: Random) => prng.nextDouble * t}.count()
```

Comment:
note that the tests will fail because these closures are expected to have `return` statements in them. Can you model these closures after the other ones in this test?